src/viewport.js

```diff
@@ -60,6 +38,34 @@ Viewport.prototype = {
     return { width, height, positionX, positionY, directionX, directionY }
   },
 
+  setup(handleScrollResize) {
+    let scheduled = false
+    const _handler = () => {
+      if (!scheduled) {
+        scheduled = true
+        requestAnimationFrame(() => {
+          const { observers } = this
+          const state = this.getState()
+          for (let i = observers.length; i--; ) observers[i].check(state)
+          this.lastX = state.positionX
+          this.lastY = state.positionY
+          scheduled = false
+        })
+      }
+    }
+    this.originalHandler = _handler
+    this.handler = handleScrollResize ? handleScrollResize(_handler) : _handler
+
+    const { handler, originalHandler } = this
+
+    addEventListener('scroll', handler, true)
+    addEventListener('resize', handler, true)
+    addEventListener('DOMContentLoaded', () => {
+      const mutationObserver = (this.mutationObserver = new MutationObserver(originalHandler))
```

Comment:
I'd use the custom handler here too.  Also probably means handleScrollResize needs a more generic name

Reply:
I intentionally left this here since you would probably want `MutationObserver` to have default implementation (e.g. not debounced checks for mutation changes). Scroll and resize events are after all really specific to user behavior :)